### PR TITLE
Export auth NoopClient for downstream use

### PR DESCRIPTION
**Pull request by anuptalwalkar**

### CODE REVIEW

File: auth/auth_stub.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ var _ Client = &noop{}
 type noop struct {
 }
 
+// NoopClient is used for testing and no-op integration
+var NoopClient = noopClient(nil)
```
Comment (sectioneight) [inline auth/auth_stub.go]: Maybe make a group with line 25?

```go
var (
  _ Client = &noop{}

  NoopClient = noopClient(nil)
)
```

File: auth/auth_stub.go [hunk 2]
```diff
@@ -22,7 +22,12 @@ package auth
 
 import "context"
 
-var _ Client = &noop{}
+var (
+	// NoopClient is used for testing and no-op integration
+	NoopClient = noopClient(nil)
```
Comment (glibsm) [inline auth/auth_stub.go]: yarpc does `Nop`, we do `Noop`, and we also have `NullHost`....

How do we consolidate?

**Comment by sectioneight** [comment]
I think Nop is the correct term from the stdlib (e.g. ioutil.NopCloser). We
should clean up.

File: auth/auth_stub.go (hunk 2 above)
Comment (madhuravi) [inline auth/auth_stub.go]: +1 This has been bothering me as well. We need to decide on a convention for uberfx atleast.

Comment (anuptalwalkar) [inline auth/auth_stub.go]: Nullop? jk, consolidating to Nop sounds good to me.

Comment (anuptalwalkar) [inline auth/auth_stub.go]: https://github.com/uber-go/fx/pull/164

**Pull request merged**